decidim-admin/app/forms/decidim/admin/attachment_form.rb

```diff
@@ -10,11 +10,26 @@ class AttachmentForm < Form
       attribute :file
       translatable_attribute :title, String
       translatable_attribute :description, String
+      attribute :weight, Integer, default: 0
+      attribute :attachment_collection_id, Integer
 
       mimic :attachment
 
       validates :file, presence: true, unless: :persisted?
       validates :title, :description, translatable_presence: true
+      validates :attachment_collection_id, inclusion: { in: :attachment_collection_ids }, allow_blank: true
```

Comment:
I'd also add a `attachment_collection` that finds the collection by its ID to make sure it exists (and add it as a presence validation)